### PR TITLE
Add support for OpenBSD sndio audio subsystem

### PR DESCRIPTION
**Pull request by ledeuns**
This PR adds support for OpenBSD sndio audio subsystem.

> $ xfreerdp /v:rdp-server /u:user /sound:sys:sndio

will enable sound on the remote machine.

Original diff by Ingo Feinerer on OpenBSD mailing-list : https://marc.info/?l=openbsd-ports&m=159974919419510&w=2

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

File: winpr/libwinpr/CMakeLists.txt [hunk 1]
```diff
@@ -77,11 +81,14 @@ macro (winpr_definition_add)
      set (WINPR_DEFINITIONS ${WINPR_DEFINITIONS} PARENT_SCOPE)
 endmacro()
 
+if (!OpenBSD)
```
Comment (akallabeth) [inline winpr/libwinpr/CMakeLists.txt:84]: is this syntax supported by `CMake` ? I think this should be `if (NOT OpenBSD)`

File: CMakeLists.txt [hunk 2]
```diff
@@ -194,21 +194,24 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "BSD")
...
 endif()
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "DragonFly")
 	set(BSD TRUE)
 	set(FREEBSD TRUE)
+	set(EPOLLSHIM TRUE)
 endif()
 
-if(FREEBSD)
+if(EPOLLSHIM)
```
Comment (akallabeth) [inline CMakeLists.txt:214]: Where is ` EPOLLSHIM` defined? I can only find a `Find` module when searching for this, but not default in ` CMake`

**Review by akallabeth — CHANGES_REQUESTED** [review]
Thank you, a really nice addition!
Please check the comments, when resolved this can be merged.

Please also run `clang-format` on the changes, the pull request tester checks the formatting style for new contributions.

**Comment by ledeuns** [comment]
Th PR is mixed with another diff. Closing and will reopen with only changes needed for sndio

**Pull request closed without merging**